### DISCUSSION TITLE
Does Chakra UI really need framer-motion?

### DISCUSSION THREAD

**Discussion by nachogoncam** [chakra-ui/chakra-ui#7285] [General]
Chakra UI uses framer-motion under the hood to animate some of its components. And Framer is a great library, don't get me wrong, but it adds 32kb (**gzipped!!**) to the bundle.

That's just huge. Especially because the animations in Chakra are simple enough to use a smaller library.

In my opinion, Framer is a bit of an overkill for this use case.

There's also no way of opting out of it. Not even if you don't use any of the animated components that depend on Framer, it's still added as a dependency of chakra/react

**Comment by campmedia** [comment]
I fully agree I'm doing some tree shaking and discovered this same thing. It should certainly be build out with an opt-out option. 32kb is a lot when dealing with first load page speed. Disappointing that a good framework would choose this route.

Take the poll https://github.com/chakra-ui/chakra-ui/discussions/8377
